test/autorest/complexgroup/array_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package complexgrouptest
+
+import (
+	"context"
+	"generatortests/autorest/generated/complexgroup"
+	"net/http"
+	"testing"
+)
+
+func getArrayOperations(t *testing.T) complexgroup.ArrayOperations {
+	client, err := complexgroup.NewDefaultClient(nil)
+	if err != nil {
+		t.Fatalf("failed to create complex client: %v", err)
+	}
+	return client.ArrayOperations()
+}
+
+func TestArrayGetEmpty(t *testing.T) {
+	client := getArrayOperations(t)
+	result, err := client.GetEmpty(context.Background())
+	if err != nil {
+		t.Fatalf("GetEmpty: %v", err)
+	}
+	val := complexgroup.ArrayWrapper{Array: []string{}}
+	expected := &complexgroup.ArrayGetEmptyResponse{
+		StatusCode:   http.StatusOK,
+		ArrayWrapper: &val,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestArrayGetNotProvided(t *testing.T) {
+	client := getArrayOperations(t)
+	result, err := client.GetNotProvided(context.Background())
+	if err != nil {
+		t.Fatalf("GetNotProvided: %v", err)
+	}
+	val := complexgroup.ArrayWrapper{}
+	expected := &complexgroup.ArrayGetNotProvidedResponse{
+		StatusCode:   http.StatusOK,
+		ArrayWrapper: &val,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestArrayGetValid(t *testing.T) {
+	client := getArrayOperations(t)
+	result, err := client.GetValid(context.Background())
+	if err != nil {
+		t.Fatalf("GetValid: %v", err)
+	}
+	val := complexgroup.ArrayWrapper{Array: []string{"1, 2, 3, 4", "", "", "&S#$(*Y", "The quick brown fox jumps over the lazy dog"}}
+	expected := &complexgroup.ArrayGetValidResponse{
+		StatusCode:   http.StatusOK,
+		ArrayWrapper: &val,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+// func TestArrayPutEmpty(t *testing.T) {
+// 	client := getArrayOperations(t)
+// 	result, err := client.PutEmpty(context.Background(), complexgroup.ArrayWrapper{Array: []string{}})
+// 	if err != nil {
+// 		t.Fatalf("PutEmpty: %v", err)
+// 	}
+// 	expected := &complexgroup.ArrayPutEmptyResponse{
+// 		StatusCode: http.StatusOK,
+// 	}
+// 	deepEqualOrFatal(t, result, expected)
+// }
+
+// func TestArrayPutValid(t *testing.T) {
+// 	client := getArrayOperations(t)
+// 	result, err := client.PutValid(context.Background(), complexgroup.ArrayWrapper{Array: []string{"1, 2, 3, 4", "", "", "&S#$(*Y", "The quick brown fox jumps over the lazy dog"}})
```

Comment:
The endpoint is expecting a null string value but sending a "" doesn't seem to work. Again, any ideas on how to send a null string value?

Reply:
Also, so you dont have to look it up, in track1 we send it over like this: 
`array := []string{"1, 2, 3, 4", "", "", "&S#$(*Y", "The quick brown fox jumps over the lazy dog"}`

Reply:
Added an issue to update []string to []*string here: #248. Leaving this commented until we add that.